libbeat/outputs/fileout/config.go

```diff
@@ -14,3 +20,16 @@ var (
 		RotateEveryKb: 10 * 1024,
 	}
 )
+
+func (c *config) Validate() error {
+	if c.Filename == "" && c.Index == "" {
```

Comment:
I would also consider it an error to set both `filename` and `index`.

I had no idea that "index" was an option for fileout. It's not in the docs.